python/cudf/cudf/core/dtypes.py

```diff
@@ -39,6 +39,26 @@
     from cudf.core.buffer import Buffer
 
 
+DTYPE_ENUM_MAP = {
+    PANDAS_NUMPY_DTYPE: 1,
+    pd.core.dtypes.dtypes.ArrowDtype: 2,
+    pd.core.dtypes.dtypes.ExtensionDtype: 3,
```

Comment:
IMO I don't think we should be supporting any arbitrary `ExtensionDtype` at this point - this would include 3rd party custom `ExtensionDtype`